scylla/src/transport/session.rs

```diff
@@ -1298,6 +1328,7 @@ impl Session {
 
         let runner = async {
             let cluster_data = self.cluster.get_data();
+            self.load_balancer.update_cluster_data(&cluster_data);
```

Comment:
You need to add a similar line in `iterator.rs` as well. It can be placed somewhere at the beginning of the `RowIteratorWorker::work()` function.

As a side note, we should probably try to think about refactoring `RowIteratorWorker::work` and `Session::execute_query` into common code... They are a bit similar and also complicated at the same time, and it's easy to miss one of them when modifying the other. OFC, it's not something that should be done in this PR.

Reply:
Latency measurements should be updated in `iterator.rs` if possible, too.